app/views/posts/index.html.erb

```diff
@@ -1 +1,33 @@
-<h1>Index for all posts</h1>
\ No newline at end of file
+<div class="card" style="width: 30rem;">
+  <div class="user-details container fluid d-flex flex-row border-bottom">
+    <img src='<%= @user.photo %>' class="card-img-top" id="user-photo" alt="walk">
+    <div class="card-body d-flex flex-column">
+      <h3 class="card-title"><%= @user.name %></h3>
+      <p class="card-text text-end"><small class="text-body-secondary">Number of posts: <%= @user.posts_counter %></small></p>
+    </div>
+  </div>
+
+  <ul class="list-group list-group-flush" id="all-posts">
+    <% i=1 %>
+    <% @user.posts.each do |p| %>     
+      <li class="list-group-item">
+        <h5>Post <%= i %>: <strong><%= p.title %></strong></h5>
+      </li>
+      <% i += 1 %>
+      <li class="list-group-item"><%= p.text.truncate(90) %></li>
+      <li class="list-group-item d-flex flex-row" id="lks-posts">
+        <%= link_to 'View post', user_post_path(@user, p), class: 'btn btn-view' %>
+        <p>Comments: <%= p.comments_counter %></p>
+        <p>Likes: <%= p.likes_counter %></p>
+      </li>
+      <li>
+        <% p.last_five_comments.each do |c| %>
+            <p><strong><%= c.user.name %></strong>: <%= c.text %></p>
+        <% end %>
```

Comment:
- [ ] [OPTIONAL] Here, I changed the variable name from c to comment. This makes the code more self-explanatory, and someone reading the code can immediately understand that the variable represents an individual comment. Descriptive variable names contribute to code clarity, especially when revisiting the code or when others are reading it.

``` ruby
<% p.last_five_comments.each do |comment| %>
  <p><strong><%= comment.user.name %></strong>: <%= comment.text %></p>
<% end %>
```